src/foam/dao/AbstractDAO.js

```diff
@@ -486,18 +486,19 @@ return this.select_(this.getX(), sink, 0, this.MAX_SAFE_INTEGER, null, null);
 
     {
       name: 'find',
-      code: function find(id) {
+      code: async function find(id) {
         // Temporary until DAO supports find_(Predicate) directly
         if ( foam.mlang.predicate.Predicate.isInstance(id) ) {
+          return (await this.where(id).limit(1).select()).array[0] ?? null;
```

Comment:
Does this break if the array's length is 0?

Reply:
no reading beyond the end of an array just returns undefined in js

Reply:
the ?? will turn undefined into null so this should match the existing API contraact

Reply:
JS still amazes me after all these years. I'm not sure that's a good thing.